Sources/NIO/ChannelOption.swift

```diff
@@ -78,6 +85,13 @@ public enum AllocatorOption: ChannelOption {
     public typealias OptionType = ByteBufferAllocator
 
     case const(())
+
+    public static func == (lhs: AllocatorOption, rhs: AllocatorOption) -> Bool {
```

Comment:
Can these trivial ones not be synthesised?

Reply:
@Lukasa unfortunately it doesn't do it, even when explicitly adding `, Equatable`. maybe the associated types?

Reply:
Hrm. @lorentey do you have any idea why this wouldn't synthesise?

Reply:
it's https://bugs.swift.org/browse/SR-9925

Reply:
Yes, sadly tuple types (including `Void`) aren't `Equatable`/`Hashable` yet. 

One workaround is to replace `()` with a nominal unit type. A dummy empty struct or single-element enum can be trivially `Equatable`/`Hashable`, and it would allow synthesis for `AllocatorOption`, too.

Reply:
@lorentey thanks! No idea why we actually has `case const(())`, maybe that was some workaround for some pre-Swift 4 bug? Who knows, your suggestion works now which is all that matters. Thank you!